### PR TITLE
ON-425 Do not check if track new for magento <2.3.1

### PR DESCRIPTION
**Pull request by vitaliyreznikov**
We use some trick to know if the track just created or not.

If the track isn't just created we do not send it to the server, because it means we already sent it before.
But this trick works for Magento >=2.3.1 only
Before this comment, https://github.com/magento/magento2/commit/2307e16105e031c0826d218cf81ffe7949191a0b Magento in `Sales/Model/ResourceModel/Order/Shipment/Relation::processRelation` uses trackcollection to go over tracks when it saves shipment.
Trackcollection updates all tracks when it getIterator, so we have no way to know if the object from observer is new or not.
In this case, we can just send the track to bolt, it's not a very big deal, actually bolt server does nothing when it receives a duplicate track.

Fixes: [ON-425](https://boltpay.atlassian.net/browse/ON-425)

#changelog Do not check if track new for magento <2.3.1

# Type of change

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

**Review by bohanzhang-bolt — APPROVED** [review]

File: Observer/TrackingSaveObserver.php [hunk 1]
```diff
@@ -125,6 +126,28 @@ private function flatten($attribute) {
         return $attribute;
     }
 
+    /**
+     * @param Track $observer
+     */
```
Comment (bohanzhang-bolt) [inline Observer/TrackingSaveObserver.php:131]: missing a @return type?

**Review by meganschnedar-bolt — APPROVED** [review]
@vitaliyreznikov I tested these changes out on Buywake's staging environment and it worked as expected (YAY). Overall, LGTM

**Comment by danac-gs** [comment]
please fix the tests

**Comment by adentong** [comment]
Since I introduced the merge conflicts, I fixed it and also the tests.

**Comment by vitaliyreznikov** [comment]
Thanks a lot @adentong ! Really appreciate it

**Review by njfaries — APPROVED** [review]

**Pull request merged**